### ISSUE TITLE
C99 "compound literals are not allowed" is shown sometimes

### ISSUE THREAD

**Issue by Yanpas** [microsoft/vscode-cpptools#1313]
* Operating System : Linux Ubuntu
* VS Code version 1.18.1
* C/C++ extension version 0.14.3

Sampel code:
```
struct Foo {
	int a,b;
};

void foo(struct Foo* f) {
	//struct Foo f = {.a = 1};
	foo(&(struct Foo){.a=1});
        /* Sometimes " a compound literal of type "struct Foo" is not allowed" is shown */
}
```

**Comment by sean-mcmanus** [comment]
The same error repros with VS 2017. I've filed a bug with the team responsible.

**Comment by sean-mcmanus** [comment]
I've verified this is fixed in our pending February update.

**Comment by sean-mcmanus** [comment]
Fixed with the preview .vsix at https://github.com/Microsoft/vscode-cpptools/releases/tag/v0.15.0-insiders (our final release is planned for next week).

**Comment by sean-mcmanus** [comment]
@Yanpas The bogus `compound literals are not allowed` error has been fixed with 0.15.0.
